### DISCUSSION TITLE
Setting breakpoints on conditional instructions

### DISCUSSION THREAD

**Discussion by kivancxxx** [x64dbg/x64dbg#2976] [Q&A]
I wonder if there is a method to set a breakpoint on every conditional state (ect... test eax, eax) in x64dbg?

Thank you so much...

**Comment by mrexodia** [comment]
Not exactly. There are basically two options (and both require custom code):

1. Disassemble everything and set breakpoints on all test/cmp instructions
2. Use tracing with a condition based on an expression function that returns whether something is a “conditional” instruction

Both are not ideal. What do you want to achieve with this functionality?

**Reply by kivancxxx** [reply]
With this functionality, I want to set breakpoints on every test eax, eax instructions since x64dbg doesn't give me the string calls or the intermodular calls to set breakpoints.

**Reply by mrexodia** [reply]
I don’t understand what you mean. What are “string calls”?

**Reply by kivancxxx** [reply]
I can't find strings/callers of function by viewing string references window or by using intermodular calls method.

**Reply by matheus-1618** [reply]
I think he means  strings that can be used as args in exec funtions such as execvp in Posix.

**Comment by HenryABM** [comment]
I think the debugger that has this functionality of automatically putting breakpoint on all conditionals will certainly be ahead of any other debugger tool.

Sometimes we want to debug all the code from conditional tests when we don't find strings or intermodular calls in the binary in question.

**Reply by mrexodia** [reply]
As answered before, it’s out of scope of x64dbg’s core functionality. You can create a script to do it exactly in the way you want.
